src/main/java/net/minecraftforge/recipe/RecipeManager.java

```diff
@@ -0,0 +1,74 @@
+package net.minecraftforge.recipe;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import net.minecraft.item.ItemStack;
+
+/**
+ * This class provides methods that are used to register machines and add recipes.
+ * 
+ * To reduce the number of HashMap.get() calls necessary, this class keeps track of the
+ * last RecipeHandler that was used, activeRecipeHandler. This reference changes every
+ * time a new machine is registered or setRecipesForMachine() is called.
+ * 
+ * @author Nephroid
+ */
+public class RecipeManager
+{
+    private static HashMap<String, RecipeHandler> machineLookupTable = new HashMap<String, RecipeHandler>();;
+    private static RecipeHandler activeRecipeHandler;
+
+    public static void registerRecipeMachine(IRecipeMachine machine)
+    {
+        if (machineLookupTable.containsKey(machine.getUniqueMachineName()))
+        {
+            System.out.format("[WARNING] Another machine with name %s has already been registered.\n Skipping this one.\n", machine.getUniqueMachineName());
```

Comment:
Should give more information about this.. or just make getRecipeHandler() smarter and remove this function all together.